custom-instrumentation/instrument-datastore/simple-datastore.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+// For a datastore, there are only two types of things to record: operations and queries.
+class SimpleDatastore {
+    constructor() {
+        this.client = null;
+        this.connected = false;
+    }
+
+    // These are our datastore operation methods
+    connect() {
+        this.connected = true;
+        console.log('Connected to simple-datastore');
+    }
+
+    close() {
+        this.connected = false;
+        console.log('Closed connection to simple-datastore');
+    }
+
+    // These are our datastore query methods
+    async execute(query) {
+        return new Promise((resolve, reject) => {
+            if (!this.connected) {
+                reject('Not connected');
```

Comment:
Promise rejections should be an instance of `Error`. Also, `return` is missing:

```suggestion
                return reject(Error('Not connected'))
```